improver/ensemble_copula_coupling/ensemble_copula_coupling.py

```diff
@@ -231,6 +231,34 @@ def __init__(
         self.ecc_bounds_warning = ecc_bounds_warning
         self.skip_ecc_bounds = skip_ecc_bounds
 
+    @staticmethod
+    def _assess_if_ecc_bounds_needed(
+        original_percentiles: ndarray,
+        desired_percentiles: ndarray,
+    ) -> bool:
+        """
+        Function to check whether ECC bounds are required for the percentile generation
+        process. ECC bounds are only necessary if any of the desired percentiles are
+        greater than/less than the largest/smallest of the original percentiles.
+
+        Args:
+            original_percentiles:
+                Array of the original percentiles.
+            desired_percentiles:
+                Array of the desired percentiles.
+
+        Returns:
+            Boolean indicating whether ECC bounds are needed (True if needed, False if
+            not).
+        """
+        if (
+            desired_percentiles[0] < original_percentiles[0]
+            or desired_percentiles[-1] > original_percentiles[-1]
+        ):
+            return True
+        else:
+            return False
```

Comment:
This could be a one-liner and is still just as readable (IMO)
```suggestion
        return (
            desired_percentiles[0] < original_percentiles[0]
            or desired_percentiles[-1] > original_percentiles[-1]
        )
```